src/dos/cdrom_image.cpp

```diff
@@ -558,33 +870,38 @@ bool CDROM_Interface_Image::LoadCueSheet(char *cuefile)
 			|| command == "PERFORMER" || command == "POSTGAP" || command == "REM"
 			|| command == "SONGWRITER" || command == "TITLE" || command == "") success = true;
 		// failure
-		else success = false;
-
+		else {
+			delete track.file;
+			track.file = NULL;
+			success = false;
+		}
 		if (!success) return false;
 	}
 	// add last track
 	if (!AddTrack(track, shift, prestart, totalPregap, currPregap)) return false;
-	
+
 	// add leadout track
 	track.number++;
 	track.attr = 0;//sync with load iso
 	track.start = 0;
 	track.length = 0;
 	track.file = NULL;
-	if(!AddTrack(track, shift, 0, totalPregap, 0)) return false;
+	if(!AddTrack(track, shift, -1, totalPregap, 0)) return false;
 
 	return true;
 }
 
+
+
 bool CDROM_Interface_Image::AddTrack(Track &curr, int &shift, int prestart, int &totalPregap, int currPregap)
 {
 	// frames between index 0(prestart) and 1(curr.start) must be skipped
 	int skip;
-	if (prestart > 0) {
+	if (prestart >= 0) {
 		if (prestart > curr.start) return false;
 		skip = curr.start - prestart;
 	} else skip = 0;
```

Comment:
`if` got touched, then let's fix formatting in line 903:

    } else {
        skip = 0;
    }

Alternatively, `skip` can be initialized to 0, making `else` unnecessary.

Reply:
Fixed in bbc159a9.